src/OrchardCore.Build/Dependencies.AspNetCore.props

```diff
@@ -1,10 +1,32 @@
 <Project>
   <PropertyGroup>
-    <AspNetCoreVersion>3.1.11</AspNetCoreVersion>
-    <AspNetCoreTargetFramework>netcoreapp3.1</AspNetCoreTargetFramework>
+    <AspNetCoreTargetFrameworks>netcoreapp3.1;net5.0</AspNetCoreTargetFrameworks>
+    <TargetFrameworks>$(AspNetCoreTargetFrameworks)</TargetFrameworks>
   </PropertyGroup>
+  <Choose>
+    <When Condition=" '$(TargetFramework)' == 'net5.0' ">
+      <PropertyGroup>
+        <AspNetCoreVersion>5.0.1</AspNetCoreVersion>
+      </PropertyGroup>
+    </When>
+    <Otherwise>
+      <!--
+      Fallback when not targetting .net 5.0.
+      At this time, that would mean .net core 3.1.  If there are more target frameworks in the future,
+      this logic should change slightly.
+      -->
+      <PropertyGroup>
+        <AspNetCoreVersion>3.1.11</AspNetCoreVersion>
+      </PropertyGroup>
+    </Otherwise>
+  </Choose>
+
+  <ItemGroup>
+    <!-- Needs to stay at older package due to items being obsoleted -->
+    <PackageManagement Include="Microsoft.AspNetCore.Authentication.AzureAD.UI" Version="3.1.11" />
```

Comment:
Can you give more details? We need to track this, and eventually bump the version.

Reply:
I saw the comment on the PR description now. We still need to find a solution. Even if that means `#if` statements in the code.

Reply:
@sebastienros The following warnings occur when using the latest package:

```
Build succeeded.

/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs(18,32): warning CS0618: 'AzureADOptions' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs(53,44): warning CS0618: 'AzureADOptions' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs(74,31): warning CS0618: 'AzureADOptions' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/OpenIdConnectOptionsConfiguration.cs(12,42): warning CS0618: 'AzureADOptions' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/OpenIdConnectOptionsConfiguration.cs(15,66): warning CS0618: 'AzureADOptions' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs(40,31): warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs(46,31): warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/CookieOptionsConfiguration.cs(25,61): warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/CookieOptionsConfiguration.cs(26,63): warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/OpenIdConnectOptionsConfiguration.cs(23,25): warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/OpenIdConnectOptionsConfiguration.cs(28,54): warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Startup.cs(60,63): warning CS0618: 'AzureADOptions' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs(55,38): warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs(78,38): warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs(83,40): warning CS0618: 'AzureADDefaults' is obsolete: 'This is obsolete and will be removed in a future version. Use Microsoft.Identity.Web instead. See https://aka.ms/ms-identity-web.' [/Users/df/development/icagile/dafergu2/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/OrchardCore.Microsoft.Authentication.csproj]
    15 Warning(s)
```
I figured the safest route would be to simply use the older version rather than try and include an update to the Azure AD code for this PR.  Would it be reasonable to use the older version in this PR and create an issue to make sure this gets upgraded soon?  Or, would it be reasonable to use the newer version and disable the warnings for now?

Reply:
Instead please just add an ignore warnings pragma in the AzureADOptionsConfiguration file

Reply:
@sebastienros I've added pragma warning disable/restores in the impacted files in the latest commit